docs-en/chapter_stack_and_queue/queue.md

```diff
@@ -357,15 +357,15 @@ Both enqueue and dequeue operations only require a single operation, each with a
 === "pop()"
     ![array_queue_pop](queue.assets/array_queue_pop.png)
 
-You might notice a problem: as enqueue and dequeue operations are continuously performed, both `front` and `rear` move to the right and **will eventually reach the end of the array and can't move further**. To resolve this issue, we can treat the array as a "circular array."
+You might notice a problem: as enqueue and dequeue operations are continuously performed, both `front` and `rear` move to the right and **will eventually reach the end of the array and can't move further**. To resolve this, we can treat the array as a "circular array" where connecting the end of the array back to its beginning.
 
-For a circular array, `front` or `rear` needs to loop back to the start of the array upon reaching the end. This cyclical pattern can be achieved with a "modulo operation," as shown in the code below:
+In a circular array, `front` or `rear` needs to loop back to the start of the array upon reaching the end. This cyclical pattern can be achieved with a "modulo operation" as shown in the code below:
 
 ```src
 [file]{array_queue}-[class]{array_queue}-[func]{}
 ```
 
-The above implementation of the queue still has limitations: its length is fixed. However, this issue is not difficult to resolve. We can replace the array with a dynamic array to introduce an expansion mechanism. Interested readers can try to implement this themselves.
+The above implementation of the queue still has its limitations: its length is fixed. However, this issue is not difficult to resolve. We can replace the array with a dynamic array that can resize itself. Interested readers can try to implement this themselves.
```

Comment:
"resize itself" may not be an accurate translation to Chinese wording. Suggestion: "dynamic array that can expand itself if needed"